File: backend/app/Models/MedicalRecord.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class MedicalRecord extends Model
{
    use HasFactory;

    // The name of the table
    protected $table = 'medical_records';

    // The primary key for the table
    protected $primaryKey = 'record_id';

    // Mass assignable fields
    protected $fillable = [
        'patient_id',
        'appointment_id',
        'diagnosis',
        'treatment_plan',
        'note',
        'status',
        'created_by',
        'updated_by',
        'attachments',
    ];

    // Relationships
    public function patient()
    {
        return $this->belongsTo(Patient::class, 'patient_id');
    }

    public function appointment()
    {
        return $this->belongsTo(Appointment::class, 'appointment_id');
    }

    // Define status options
    const STATUS_ACTIVE = 'Active';
    const STATUS_ARCHIVED = 'Archived';
    const STATUS_INACTIVE = 'Inactive';

    // Define created and updated at fields
    const CREATED_AT = 'created_at';
    const UPDATED_AT = 'updated_at';

    // Set up automatic handling of timestamps
    public $timestamps = true;
}

File: backend/app/models/medical_record.py
#app/models/medical_record.py

from sqlalchemy import Column, Integer, String, DateTime, ForeignKey, Text
from sqlalchemy.orm import relationship
from datetime import datetime

from .base import Base

class MedicalRecord(Base):
    __tablename__ = 'medical_records'

    id = Column(Integer, primary_key=True, index=True)
    patient_id = Column(Integer, ForeignKey('patients.id'), nullable=False)
    doctor_id = Column(Integer, ForeignKey('doctors.id'), nullable=False)
    appointment_id = Column(Integer, ForeignKey('appointments.id'), nullable=False)

    diagnosis = Column(String(255), nullable=False)
    symptoms = Column(Text, nullable=False)
    treatment = Column(Text, nullable=False)
    record_type = Column(
        String(100), 
        default= datetime.utcnow, 
        onupdate=datetime.utcnow
    )

    record_date = Column(DateTime, default=datetime.utcnow)
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
    
    # Relationships
    patient = relationship("Patient", back_populates="medical_records")
    doctor = relationship("Doctor", back_populates="medical_records")
    appointment = relationship("Appointment", back_populates="medical_records", uselist=False)
    
    def __repr__(self):
        return f"<MedicalRecord(id={self.id}, appointment_id={self.appointment_id}, record_date={self.record_date})>"
    